executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -347,14 +396,34 @@ impl<'a, T: FieldElement> Machine<'a, T> for FixedLookup<'a, T> {
             .peekable();
 
         let outer_query = OuterQuery::new(caller_rows, identity);
-        self.process_plookup_internal(caller_rows, &outer_query.left, right)
+        self.process_plookup_internal(caller_rows, &outer_query.left, right, identity_id)
     }
 
     fn take_witness_col_values<'b, Q: QueryCallback<T>>(
         &mut self,
         _mutable_state: &'b mut MutableState<'a, 'b, T, Q>,
     ) -> HashMap<String, Vec<T>> {
-        Default::default()
+        let mut witness_col_values = HashMap::new();
+        if self.logup_multiplicity_column.is_some() {
+            assert!(
+                self.multiplicities.len() <= 1,
+                "LogUp witness generation not yet supported for > 1 lookups"
+            );
+            log::trace!("Detected LogUp Multiplicity Column");
+            for multiplicity in std::mem::take(&mut self.multiplicities).into_values() {
+                let mut values = vec![];
+                for row in 0..self.degree as usize {
+                    values.push(T::from(multiplicity.get(&row).cloned().unwrap_or_default()));
```

Comment:
Hm, do we really want to merge it like this? Why not create `self.multiplicities` as a vector of size `self.degree` from the start? It would be so much faster, I think.

Reply:
And it should already be a `T` - or do you think one conversion from integer to field element is cheaper than incrementing a field element for each access?

Reply:
Converting is more expensive, you're right. Now, fixed it